common/overlaysynccleanuphandler.h

```diff
@@ -43,6 +43,17 @@ class OverlaySyncCleanupHandler : public CleanUpHandler
      * @return ReturnCode::FAILURE on failure
      */
     ReturnCode clean() override;
+
+    /**
+     * @brief queryName is needed to query member name yet its full functionality is not needed for
+     * this class.
+     *
+     * Since OverlaySyncCleanupHandler uses DirectoryCleanUpHandler for removing directory, this
+     * function won't be called for the same reason.
+     *
+     * @return an empty string
+     */
+    const std::string queryName() override;
```

Comment:
Actually we are returning empty string because no directory removal operation occurs within OverlaySyncCleanupHandler, so it is irrelevant when the code checks whether the directory is in the list or not since we are doing this for cleaning path. Maybe i should add this as a comment.